### PR TITLE
Issue #2909 - Refactor ellipsis menu containers/federations

### LINKED ISSUE #2909: Refactor ellipsis menu containers/federations
### Description
The ellipsismenu has been implemented by creating a `<EllipsisButtonWithMenu />` with a list as prop. I think that this is not a very good approach, the developer must have extra knowledge for the format of this list and is not very component oriented. Refactoring to a more react-like easy to use would enhance readability and componentization. 

### Goals
- [ ] Provide the same functionality as before but with a more  react approach
- [ ] Replace it in containers and federations

### Tasks
- [ ] separate `<EllipsisButtonWithMenu  >` into `<EllipsisMenu  />` and `<EllipsisMenuItem />`
- [ ]  implement the lists for containers
- [ ]  implement the lists for federations

### Details
Example of  reimplementation `<EllipsisContainerMenu />`:

-  In  `src\v5\ui\routes\dashboard\projects\containers\containersList\containerListItem\containerListItem.component.tsx` it should look this way:
```
...
			<DashboardListItemIcon selected={isSelected}>
				<EllipsisContainerMenu  container={container}/>
			</DashboardListItemIcon>
...
```
- `<EllipsisContainerMenu />` (this would be a new component) would look this way:
```
const ContainerEllipsisMenu =  (container:IContainer) => {
  <EllipsisMenu>
    <EllipsisMenuItem onClick="...">
      <FormattedMessage id="containers.ellipsisMenu.loadContainer" defaultMessage="Load Container in 3D Viewer" />
    <EllipsisMenuItem />
         ....  
  <EllipsisMenu>
...
```

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #2909 

#### Description
- Ellipsis menu is now a React component that can be re-used between federations and containers; 
- The functionalities are kept as before.

### CODE REVIEW

File: frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { ReactNode, SyntheticEvent } from 'react';
+import { Typography } from '@material-ui/core';
+import { Link } from 'react-router-dom';
+import { MenuItem } from './ellipsisMenuItem.styles';
+
+type EllipsisMenuItemLinkProps = {
+	title: ReactNode;
+	to: string;
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx:24]: Is it posible to make the to optional also?
In that case lets get rid of the 'to' and onclick that leads nowhere in containerellipsismenu and federationellipsismenu

File: frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	key?: string;
+	onClick?: (event: SyntheticEvent) => void;
+};
+
+export const EllipsisMenuItemLink = ({ to, title, key, onClick }: EllipsisMenuItemLinkProps) => (
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx:29]: I think I didn't communicate well why I wanted an 'EllipsisMenuItemLink'. I meant to have an 'EllipsisMenuItemLink' for when you have the { to } prop as a link, but as we only have one component  to handle both cases lets change this one back to  'EllipsisMenuItem'

File: frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containerListItem/containerEllipsisMenu/containerEllipsisMenu.component.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				to=""
+				onClick={openShareModal}
+			/>
+			<EllipsisMenuItemLink
+				title={formatMessage({
+					id: 'containers.ellipsisMenu.settings',
+					defaultMessage: 'Settings',
+				})}
+				to=""
+				onClick={() => { }}
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containerListItem/containerEllipsisMenu/containerEllipsisMenu.component.tsx:115]: lets get rid of the empty 'tos' and 'onclick'

File: frontend/src/v5/ui/routes/dashboard/projects/federations/federationsList/federationListItem/federationEllipsisMenu/federationEllipsisMenu.component.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+				onClick={() => { }}
+			/>
+
+			<EllipsisMenuItemLink
+				title={formatMessage({
+					id: 'federations.ellipsisMenu.edit',
+					defaultMessage: 'Edit Federation',
+				})}
+				to=""
+				onClick={() => { }}
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/federations/federationsList/federationListItem/federationEllipsisMenu/federationEllipsisMenu.component.tsx:54]: lets get rid of the empty tos and onclick if posible

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx (hunk 2 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/ellipsisMenu/ellipsisMenuItem/ellipsisMenutItem.component.tsx]: done

File: frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containerListItem/containerEllipsisMenu/containerEllipsisMenu.component.tsx (hunk 3 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/containerListItem/containerEllipsisMenu/containerEllipsisMenu.component.tsx]: done

File: frontend/src/v5/ui/routes/dashboard/projects/federations/federationsList/federationListItem/federationEllipsisMenu/federationEllipsisMenu.component.tsx (hunk 4 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/federations/federationsList/federationListItem/federationEllipsisMenu/federationEllipsisMenu.component.tsx]: done

**Pull request merged**